### PR TITLE
Update "upcomingLeg" field to enable filter out legs by guarantee

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
We would like to ask for most upcoming leg & take guarantee type into account too.

- without argument - just get an upcoming leg to retrieve easily info about nearest flight
- with argument guarantee=KIWICOM - to get nearest flight covered by our guarantee to know whether we should display additional support

### CODE REVIEW

**Comment by jaroslav-kubicek** [comment]
Hola @mrtnzlml & @tbergq could I ask you for CR and approval? I need it to get merged today and everyone but you left me today :)

**Comment by tbergq** [comment]
I will have a look at it when I get to the office, unless @mrtnzlml beats me to it.

File: src/booking/types/outputs/BookingInterface.js [hunk 1]
```diff
@@ -307,8 +308,16 @@ export const commonFields = {
...
-      findUpcomingLeg(legs),
+    args: {
+      guarantee: {
+        type: GraphQLCoveredBy,
+        description: 'Filter out only legs with specified guarantee.',
+      },
+    },
+    resolve: (
+      { legs }: BookingInterfaceData,
+      { guarantee }: Object,
```
Comment (tbergq) [inline src/booking/types/outputs/BookingInterface.js]: Could you add a better type here?

Comment (jaroslav-kubicek) [reply src/booking/types/outputs/BookingInterface.js]: should be fixed

**Pull request merged**